iib/workers/api_utils.py

```diff
@@ -150,6 +150,11 @@ def update_request(
     # Prevent a circular import
     start_time = time.time()
     request_url = f'{config.iib_api_url.rstrip("/")}/builds/{request_id}'
+
+    log.info(f'Checking the current state of request {request_id}')
+    if requests_auth_session.get(request_url).json()['state'] in ['failed', 'complete']:
```

Comment:
I was also thinkin about importing `RequestStateMapping.get_final_states()` from `models.py`, however I did not found single example with import from `iib.web` in the `iib.workers` part. (This could possibly create circular import I believe).